src/plugins/auto/plugin.cpp

```diff
@@ -279,11 +279,11 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
                              config.first.c_str(), config.second.c_str());
                  }
              }
-             auto tmpiter = std::find_if(fullConfig.begin(), fullConfig.end(), [](const std::pair<std::string, std::string>& config) {
-                            return (config.first == CONFIG_KEY(ALLOW_AUTO_BATCHING));
-                            });
+             auto tmpiter = fullConfig.find(CONFIG_KEY(ALLOW_AUTO_BATCHING));
              if (tmpiter != fullConfig.end())
                  deviceConfig.insert({tmpiter->first, tmpiter->second});
+             if (fullConfig.find(CONFIG_KEY(AUTO_BATCH_TIMEOUT)) == fullConfig.end())  // do not override user's val
```

Comment:
1) maybe we don't need to pass this key to device all the time? like when autobatching disabled and this key was erased before in ie core, here will push the key to device config again.
2) the user's val has not been passed to device

Reply:
this PR is just a API example, pls take it, brush and change thoroughly